### PR TITLE
Remove dependencies

### PR DESCRIPTION
**Pull request by schorfES**
Hi everyone,

this PR removes 3 of the 4 dependencies of polyglot  in favour of reducing its production bundle size by ~3kb (gzipped). The implementation of the equally named functions are all inspired by their original functions but only support the required usecases for polyglot.

Thanks, Norman

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,14 +17,36 @@
...
+    fn.call(
+      receiver,
+      target[key],
+      isArray ? Number(key) : key
+    );
+  });
+}
+
+function has(target, key) {
+  return key in target;
```
Comment (ljharb) [inline index.js]: This does not check own properties, and so would massively break this library.

File: index.js [hunk 2]
```diff
@@ -17,14 +17,36 @@
...
 
-var forEach = require('for-each');
 var warning = require('warning');
-var has = require('has');
-var trim = require('string.prototype.trim');
 
-var warn = function warn(message) {
+// #### Helper methods
+function forEach(target, fn, receiver) {
+  var isArray = target instanceof Array;
```
Comment (ljharb) [inline index.js]: instanceof Array is never reliable; only Array.isArray is safe.

File: index.js [hunk 3]
```diff
@@ -17,14 +17,36 @@
...
+  });
+}
+
+function has(target, key) {
+  return key in target;
+}
+
+function trim(str) {
+  /* eslint-disable no-control-regex */
+  var whitespace = '[\x09\x0A\x0B\x0C\x0D\x20\xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u3000\u2028\u2029\uFEFF]+';
```
Comment (ljharb) [inline index.js]: this regex changes every year as a new unicode standard comes out

**Comment by schorfES** [comment]
Well that's fair enough. Just because you are the owner of all those dependencies, do you see any ways to optimize maybe some imports to reduce the bundle size of polyglot especially for browsers?

**Comment by ljharb** [comment]
My recommendation would be to use other libs that share the same dependencies, so you can dedupe.

For example, if you're already using https://npmjs.com/airbnb-browser-shims, polyglot's deps will add zero additional bundle size.

**Pull request closed without merging**